tsdate/prior.py

```diff
@@ -947,48 +972,55 @@ def gamma_cdf(t_set, alpha, beta):
     return np.insert(t_set, 0, 0)
 
 
-def fill_priors(node_parameters, timepoints, ts, Ne, *, prior_distr, progress=False):
+def make_discretised_grid(timepoints, ts):
+    """
+    Create a NodeGridValues object to containing probabilities of nodes existing in
+    a set of specified timeslices, with nodes corresponding to the input tree sequence
     """
-    Take the alpha and beta values from the node_parameters array, which contains
-    one row for each node in the TS (including fixed nodes)
-    and fill out a NodeGridValues object with the prior values from the
-    gamma or lognormal distribution with those parameters.
+    datable_nodes = np.ones(ts.num_nodes, dtype=bool)
+    datable_nodes[ts.samples()] = False
+    datable_nodes = np.where(datable_nodes)[0]
+    return base.NodeGridValues(
+        ts.num_nodes,
+        datable_nodes[np.argsort(ts.tables.nodes.time[datable_nodes])].astype(np.int32),
+        timepoints,
```

Comment:
This is the only place the time rescaling needs to be applied, I believe.

Reply:
I think that we should maintain "true" (i.e. generations, years, or whatever) time within the NodeGridValues object, because people might want to specify their own timepoints (in e.g. generations). There is also the issue that with historical samples, we will want to truncate the prior to the oldest fixed node child, where the time of the child is quoted in either years or generations.

I've put a comment in another place where I think we might need to consider the scaling.